opensrp-giz-malawi/src/main/java/org/smartregister/giz/configuration/OpdRegisterQueryProvider.java

```diff
@@ -23,17 +26,20 @@
     public String getObjectIdsQuery(@Nullable String filters, @Nullable String mainCondition) {
         Date latestValidCheckInDate = OpdLibrary.getInstance().getLatestValidCheckInDate();
         String oneDayAgo = OpdUtils.convertDate(latestValidCheckInDate, OpdDbConstants.DATE_FORMAT);
+        String todayDate = new SimpleDateFormat(GizConstants.DateTimeFormat.yyyy_MM_dd, Locale.US).format(new Date());
 
         if (!TextUtils.isEmpty(filters)) {
             if (!TextUtils.isEmpty(mainCondition)) {
                 String sql =
-                        "SELECT ec_client_search.object_id, ec_client_search.last_interacted_with, (opd_details.current_visit_start_date IS NOT NULL AND opd_details.current_visit_start_date >= '$latest_start_visit_date' AND opd_details.current_visit_end_date IS NULL) AS checked_in FROM ec_client_search WHERE ec_client_search.date_removed IS NULL AND phrase MATCH '%s*') " +
-                        "    LEFT JOIN opd_details ON ec_client_search.object_id = opd_details.base_entity_id" +
-                                "inner join client_register_type crt on crt.base_entity_id = ec_client_search.object_id  " +
-                                " WHERE crt.register_type = 'opd' AND checked_in " +
-                        "ORDER BY last_interacted_with DESC";
+                        "SELECT object_id \n" +
+                                "FROM (SELECT object_id, last_interacted_with \n" +
+                                "FROM ec_client_search \n" +
+                                "inner join client_register_type crt on crt.base_entity_id = ec_client_search.object_id  \n" +
+                                "WHERE crt.register_type = 'opd' AND  ec_client_search.date_removed IS NULL AND phrase  MATCH '%s*')\n" +
+                                "WHERE ec_client_search.object_id IN (SELECT base_entity_id\n" +
+                                "FROM opd_client_visits where visit_group = '" + todayDate + "')\n" +
+                                "ORDER BY last_interacted_with DESC";
```

Comment:
Could you modify and use join? Also is the first subquery needed?

Reply:
Removed first Subquery

Reply:
Done